Client/Frontend/Home/TopSitesPanel.swift

```diff
@@ -190,11 +195,10 @@ class TopSitesPanel: UIViewController {
         // now update the data source with the new data
         self.updateDataSourceWithSites(result)
 
-        let data = dataSource.data
         collection?.performBatchUpdates({
 
             // find out how many thumbnails, up the max for display, we can actually add
-            let numOfCellsFromData = data.count + SuggestedSites.count
+            let numOfCellsFromData = self.dataSource.topSitesCursor.count
```

Comment:
Instead of reaching into the dataSource to get the count from topSitesCursor, we should hide the implementation details of the data source and expose a count method so this can turn into `self.dataSource.count`.